.github/workflows/docker.yaml

```diff
@@ -49,43 +49,57 @@ jobs:
       with:
         authToken: '${{ secrets.CACHIX_CARDANO_SCALING_AUTH_TOKEN }}'
 
-    - name: 🔨 Build image using nix
+    - name: 🔨 🏷️ Build and tag images
       run: |
+
         IMAGE_NAME=ghcr.io/${{github.repository_owner}}/${{matrix.target}}
         echo "IMAGE_NAME=${IMAGE_NAME}" >> $GITHUB_ENV
         nix build .#docker-${{ matrix.target }}
         docker load < result
 
-        # Determine whether we are building a tag and if yes, set a VERSION_NAME
+        # Compute tags. We want the following behaviour:
+        #
+        # 1. Building on tag:      image tagged `tag`
+        # 2. Building on release:  image tagged `release`
+        # 3. Building on master:   image tagged `unstable`.
+
+        # Let's default to 'unstable'.
+        IMAGE_LABEL=unstable
+        # And the version as the git commit.
+        VERSION=${{github.sha}}
+
+        # Determine whether we are building a tag and if yes, set the label
+        # name to be the tag name, and the version to be the tag.
         BUILDING_TAG=${{github.ref_type == 'tag'}}
         [[ ${BUILDING_TAG} = true ]] && \
-          VERSION_NAME=${{github.ref_name}}
+          IMAGE_LABEL=${{github.ref_name}} && \
+          VERSION=${{github.ref_name}}
 
-        BUILDING_ON_MASTER=false
-        if (git merge-base --is-ancestor HEAD origin/master); then
-          BUILDING_ON_MASTER=true
-        fi
+        # If we're running on the release branch, tag it as `latest`,
+        # and leave the version as the git commit.
+        [[ ${GITHUB_REF_NAME} = "release" ]] && \
```

Comment:
are u sure this is going to work?

[currently](https://github.com/cardano-scaling/hydra/blob/master/.github/workflows/docker.yaml#L14) this workflow only runs on merging to master.